src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat[3];
+  const separatorTo = toFormat[3];
+
+  const dateParts = date.split(separatorFrom);
+  const dateObject = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateObject[part] = dateParts[index];
+  });
+
+  if (dateObject['YYYY']) {
+    dateObject['YY'] = dateObject['YYYY'].slice(-2);
+  } else if (dateObject['YY']) {
+    const year = parseInt(dateObject['YY'], 10);
+
+    dateObject['YYYY'] =
+      year < 30 ? '20' + dateObject['YY'] : '19' + dateObject['YY'];
+  }
```

Comment:
The conditional checks for 'YYYY' and 'YY' can be made more explicit by using 'in' operator to check for the existence of the property in the 'dateObject'. This would make the code more readable and less prone to errors if additional date parts are introduced in the future.